### ISSUE TITLE
Feature Request: checks API

### ISSUE THREAD

**Issue by mattfarina** [softprops/hubcaps#148]
Would it be possible to add support for the checks API?

https://developer.github.com/v3/checks/

**Comment by grahamc** [comment]
In my fairly out of date fork of Hubcaps without async, I have implemented support for checks, and supporting endpoints: https://github.com/grahamc/hubcaps/compare/451cf178af44995b07ac24ebc514b71c1574485a...9df8cfb966fe28e5c20eec31f548cf71b4195f5c

 - JWT auth
 - exchanging a JWT token for app install access token
 - regenerating these JWT / app tokens automatically
 - creating check results

**Comment by rocallahan** [comment]
@grahamc I've ported and reworked your branch to hubcaps master: https://github.com/rocallahan/hubcaps/tree/checks. I'm going to submit it as a PR on the assumption you won't object.

I changed a few things in your code. In particular, mutating the credentials of the Github object was worrying so I took a different approach where we pass to `post_media` constraints on what authentication method can be used. Seems cleaner and can tidily represent the constraints written in the Github API docs.
